### DISCUSSION TITLE
How to configure nginx when deploying a static site

### DISCUSSION THREAD

**Discussion by trostum** [coollabsio/coolify#3250] [Q&A]
I have a React application which is built with nixpacks and deployed as a static site using the nginx:alpine docker image.
How can i customize the nginx config?
Is it possible or do i need to switch deploy method from static site to Docker (with my own Dockerfile)?

**Comment by pushkarsingh32** [comment]
looking for the same. following if someone answers

**Comment by KianAnbarestani** [comment]
You can't customize NGINX config with the static site + nixpacks method in Coolify. To gain full control, switch to Docker deployment and write a custom `Dockerfile` that copies both your build output and your own `nginx.conf`. This gives you full flexibility over NGINX behavior.

**Comment by patriksimms** [comment]
Fortunatly, this is only partly true by now. 

You can use a custom nginx config by now. Although I do not know a way how to automatically use the nginx.conf file from the project
<img width="1186" height="339" alt="image" src="https://github.com/user-attachments/assets/1a8a7e28-16d9-43e3-be06-7f938c887983" />

**Reply by hybridherbst** [reply]
Has anyone figured this out? I think it's a bit hit-and-miss if one needs to manually take care of copying a custom nginx.conf over to Coolify.
